addon/authenticators/oauth2-password-grant.js

```diff
@@ -53,6 +53,18 @@ export default BaseAuthenticator.extend({
   */
   clientId: null,
 
+  /**
+    The client_secret to be sent to the authentication server. This should be set
+    along with the client_id if the client needs to authenticate itself against
+    the authorization server in addition to the user
+
+    @property clientSecret
+    @type String
+    @default null
+    @public
+  */
+  clientSecret: null,
```

Comment:
This was intentionally left out - as soon as you publish your client secret as part of your publicly accessible Ember app it's no longer a secret.

Reply:
You are right, I got a little over zealous, I'll remove it.